timesketch/lib/analyzers/domain.py

```diff
@@ -0,0 +1,102 @@
+"""Sketch analyzer plugin for domain."""
+from __future__ import unicode_literals
+
+import collections
+
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib import parse as urlparse  # pylint: disable=no-name-in-module
+
+from timesketch.lib import emojis
+from timesketch.lib.analyzers import interface
+from timesketch.lib.analyzers import manager
+
+
+class DomainSketchPlugin(interface.BaseSketchAnalyzer):
+    """Sketch analyzer for Domain."""
+
+    NAME = 'domain'
+
+    def __init__(self, index_name, sketch_id):
+        """Initialize The Sketch Analyzer.
+
+        Args:
+            index_name: Elasticsearch index name
+            sketch_id: Sketch ID
+        """
+        self.index_name = index_name
+        super(DomainSketchPlugin, self).__init__(index_name, sketch_id)
+
+    @staticmethod
+    def _get_domain_from_url(url):
+        """Extract domain from URL.
+
+        Args:
+            url: URL to parse.
+
+        Returns:
+            String with domain from URL.
+        """
+        # TODO: See if we can optimize this because it is rather slow.
+        domain_parsed = urlparse(url)
+        domain_full = domain_parsed.netloc
+        domain, _, _ = domain_full.partition(':')
+        return domain
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        query = (
+            '{"query": { "bool": { "should": [ '
+            '{ "exists" : { "field" : "url" }}, '
+            '{ "exists" : { "field" : "domain" }} ] } } }')
+
+        return_fields = ['domain', 'url', 'message', 'human_readable']
```

Comment:
in this case, no need for the message nor the human_readable

Reply:
good catch, fixed.